services/src/main/java/org/jboss/windup/web/services/RuleDataLoader.java

```diff
@@ -121,14 +121,29 @@ private void loadRules()
 
             for (RulesPath rulesPath : webConfiguration.getRulesPaths())
             {
-                LOG.info("Purging existing rule data for: " + rulesPath);
-                // Delete the previous ones
-                entityManager
-                            .createNamedQuery(RuleProviderEntity.DELETE_BY_RULES_PATH)
-                            .setParameter(RuleProviderEntity.RULES_PATH_PARAM, rulesPath)
-                            .executeUpdate();
+                LOG.info(String.format("Purging existing rule data for: %s", rulesPath));
+
+                logCurrentEntityCounts("1");
+                {
```

Comment:
I like them. They limit scope and reduce bug potential. Creating a method would make more mess than tidy.